### PR TITLE
Unmarshaling response

### PR DESCRIPTION
**Pull request by dscervantes**
## What?
Explain what the change is linking any relevant JIRAs or Issues.
Unmarshaling response 
https://issues.redhat.com/browse/RHCLOUD-21190

## Why?
Consider what business or engineering goal does this PR achieves.
Infrastructure are going to change the non- 200 response. We need to change the return error as an error instead of a string 

## How?
Created a new type that will send a json message 

## Testing
Did you add any tests for the change?

## Anything Else?
Any other notes about the PR that would be useful for the reviewer. 

## Secure Coding Practices Checklist Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [X ] General Coding Practices

### CODE REVIEW

File: internal/controller/account_resolver.go [hunk 1]
```diff
@@ -29,6 +29,16 @@ type AuthGwResp struct {
 	Identity string `json:"x-rh-identity"`
 }
 
+type ErrorResponse struct {
```
Comment (dehort) [inline internal/controller/account_resolver.go:32]: Can you rename this to `authGwErrorResponse`?  Its a bit wordy but it communicates its use.

In golang, starting a variable with a capital letter exports the variable out of the package.  In this case, we do not need to expose the Gateway response structures outside of this package.  We really want to hide them from the callers.  So lets not export the error response and the non-error response (please change AuthGwResp to authGwResp).

**Review by dehort — CHANGES_REQUESTED** [review]
Looking good!  I made a couple of comments.  Would you mind adding a couple of test cases to the existing tests in account_resolver_test.go?  We probably need a test for an empty json response, empty/invalid response and a valid response.

File: internal/controller/account_resolver.go [hunk 2]
```diff
@@ -83,8 +93,14 @@ func (bar *BOPAccountIdResolver) MapClientIdToAccountId(ctx context.Context, cli
 	if r.StatusCode != 200 {
 		logger.Debugf("Call to Auth Gateway returned http status code %d", r.StatusCode)
 		b, _ := ioutil.ReadAll(r.Body)
-		return "", "", "", fmt.Errorf("Unable to find account %s", string(b))
+		var errResponse ErrorResponse
+		if err := json.Unmarshal(b, &errResponse); err != nil {
+			logger.WithFields(logrus.Fields{"error": err}).Error("Unable to parse error reponse")
+			return "", "", "", err
```
[the remark this answers is not in the document]
Comment (dscervantes) [reply internal/controller/account_resolver.go:99]: I saw the %w, but didn't know I could use it like that, makes a lot of sense !
